ios/engine/KMEI/KeymanEngine/Classes/Model/KeymanPackage.swift

```diff
@@ -67,9 +58,24 @@ public class KeymanPackage
         let data = try Data(contentsOf: path, options: .mappedIfSafe)
         let jsonResult = try JSONSerialization.jsonObject(with: data, options: .mutableLeaves)
         if let jsonResult = jsonResult as? [String:AnyObject] {
-          let kmp = KeymanPackage.init(folder: folder)
-          kmp.parse(json: jsonResult)
-          return kmp
+          if let packagedKeyboards = jsonResult["keyboards"] as? [[String:AnyObject]] {
+            if let packagedModels = jsonResult["lexicalModels"] as? [[String:AnyObject]] {
+                //TODO: rrb signal error
+              return nil
+            }
+            let kmp = KeyboardKeymanPackage.init(folder: folder)
+            kmp.parse(json: jsonResult)
+            return kmp
+          }
+          else if let packagedModels = jsonResult["lexicalModels"] as? [[String:AnyObject]] {
+            if let packagedKeyboards = jsonResult["keyboards"] as? [[String:AnyObject]] {
+                //TODO: rrb signal error
```

Comment:
I assume this is planned for a later time?

Reply:
Yes. I needed to make something, since there were equivalent references to the KeyboardPicker. I had to either rip it out for now and hope to remember (ri-i-i-ight) to put it back in when we wanted to implement this,  or do it now but leave it disconnected. I chose the latter.

And yes,  we haven't defined how we want to let the user know that the .kmp file is invalid due to containing both a keyboard and a model, but here's where it should go when we decide to do it.

Reply:
I don't think we give a hugely detailed error -- show an error: `"The package $name is invalid because it contains both keyboards and lexical models"` gives support staff enough to go on.

Reply:
I've added logging an error and left a TODO note that we should show the error to the user. Is that enough? Or do I need to initiate UI from down in the bowels here...
I also got rid of a redundant check.